crates/autopilot/src/infra/solvers/dto/solve.rs

```diff
@@ -132,7 +132,65 @@ impl Solution {
 }
 
 #[serde_as]
-#[derive(Clone, Debug, Default, Deserialize)]
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+#[serde(untagged)]
+pub enum TradedOrder {
+    #[serde(rename_all = "camelCase")]
+    TradedAmount {
+        /// The effective amount that left the user's wallet including all fees.
+        #[serde_as(as = "HexOrDecimalU256")]
+        sell_amount: U256,
+        /// The effective amount the user received after all fees.
+        #[serde_as(as = "HexOrDecimalU256")]
+        buy_amount: U256,
+    },
+    #[serde(rename_all = "camelCase")]
+    TradedOrder {
+        side: Side,
+        sell_token: H160,
+        buy_token: H160,
+        #[serde_as(as = "HexOrDecimalU256")]
+        /// Sell limit order amount.
+        sell_amount: U256,
+        #[serde_as(as = "HexOrDecimalU256")]
+        /// Buy limit order amount.
+        buy_amount: U256,
+        /// The effective amount that left the user's wallet including all fees.
+        #[serde_as(as = "HexOrDecimalU256")]
+        traded_sell: U256,
+        /// The effective amount the user received after all fees.
+        #[serde_as(as = "HexOrDecimalU256")]
+        traded_buy: U256,
```

Comment:
The old enum is supposed to be removed in a follow up.
For `TradedOrder` I wanted to name the fields as they were named in the `Request`. It would be confusing to have order.sell_amount in Request while having the same field in the Response as limit_sell_amount.

Reply:
unrelated, buy maybe it's nicer to have `sold` instead of `traded_sell` and `bought` instead of `traded_buy`.

Reply:
I think they should be called `limit_sell` and `executed_sell` to make it very clear

Reply:
renamed